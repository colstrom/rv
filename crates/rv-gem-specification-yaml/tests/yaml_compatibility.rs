use rv_gem_specification_yaml::parse;
use std::fs;

fn load_fixture(name: &str) -> String {
    let fixture_path = format!("tests/fixtures/{name}.yaml");
    fs::read_to_string(&fixture_path)
        .unwrap_or_else(|_| panic!("Failed to read fixture: {fixture_path}"))
}

#[test]
fn test_parse_simple_specification() {
    let yaml_content = load_fixture("simple_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("simple_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse simple specification: {e}");
        }
    }
}

#[test]
fn test_parse_complex_specification() {
    let yaml_content = load_fixture("complex_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("complex_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse complex specification: {e}");
        }
    }
}

#[test]
fn test_parse_version_constraints_specification() {
    let yaml_content = load_fixture("version_constraints_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("version_constraints_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse version constraints specification: {e}");
        }
    }
}

#[test]
fn test_parse_minimal_specification() {
    let yaml_content = load_fixture("minimal_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("minimal_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse minimal specification: {e}");
        }
    }
}

#[test]
fn test_parse_prerelease_specification() {
    let yaml_content = load_fixture("prerelease_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("prerelease_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse prerelease specification: {e}");
        }
    }
}

#[test]
fn test_parse_licensed_specification() {
    let yaml_content = load_fixture("licensed_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("licensed_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse licensed specification: {e}");
        }
    }
}

#[test]
fn test_parse_edge_case_specification() {
    let yaml_content = load_fixture("edge_case_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("edge_case_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse edge case specification: {e}");
        }
    }
}
