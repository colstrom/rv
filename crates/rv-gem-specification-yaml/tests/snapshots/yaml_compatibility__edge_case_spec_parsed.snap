---
source: crates/rv-gem-specification-yaml/tests/yaml_compatibility.rs
expression: spec
---
Specification {
    name: "edge-case-gem",
    version: Version {
        version: "0.1.0.pre",
        segments: [
            Number(
                0,
            ),
            Number(
                1,
            ),
            Number(
                0,
            ),
            String(
                "pre",
            ),
        ],
    },
    summary: "Testing edge cases",
    require_paths: [
        "lib",
    ],
    rubygems_version: "3.4.20",
    specification_version: 4,
    date: "",
    authors: [
        "Edge Developer",
    ],
    email: [],
    homepage: None,
    description: None,
    licenses: [],
    files: [],
    executables: [],
    extensions: [],
    dependencies: [
        Dependency {
            name: "nokogiri",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: Pessimistic,
                        version: Version {
                            version: "1.13.0",
                            segments: [
                                Number(
                                    1,
                                ),
                                Number(
                                    13,
                                ),
                                Number(
                                    0,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Runtime,
        },
        Dependency {
            name: "zeitwerk",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: Pessimistic,
                        version: Version {
                            version: "2.5",
                            segments: [
                                Number(
                                    2,
                                ),
                                Number(
                                    5,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Runtime,
        },
        Dependency {
            name: "dry-types",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: GreaterEqual,
                        version: Version {
                            version: "1.5",
                            segments: [
                                Number(
                                    1,
                                ),
                                Number(
                                    5,
                                ),
                            ],
                        },
                    },
                    VersionConstraint {
                        operator: Less,
                        version: Version {
                            version: "2.0",
                            segments: [
                                Number(
                                    2,
                                ),
                                Number(
                                    0,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Runtime,
        },
        Dependency {
            name: "activesupport",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: Pessimistic,
                        version: Version {
                            version: "7.0.0",
                            segments: [
                                Number(
                                    7,
                                ),
                                Number(
                                    0,
                                ),
                                Number(
                                    0,
                                ),
                            ],
                        },
                    },
                    VersionConstraint {
                        operator: GreaterEqual,
                        version: Version {
                            version: "7.0.4",
                            segments: [
                                Number(
                                    7,
                                ),
                                Number(
                                    0,
                                ),
                                Number(
                                    4,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Runtime,
        },
        Dependency {
            name: "minitest",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: Pessimistic,
                        version: Version {
                            version: "5.0",
                            segments: [
                                Number(
                                    5,
                                ),
                                Number(
                                    0,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Development,
        },
        Dependency {
            name: "rubocop",
            requirement: Requirement {
                constraints: [
                    VersionConstraint {
                        operator: Pessimistic,
                        version: Version {
                            version: "1.50",
                            segments: [
                                Number(
                                    1,
                                ),
                                Number(
                                    50,
                                ),
                            ],
                        },
                    },
                    VersionConstraint {
                        operator: GreaterEqual,
                        version: Version {
                            version: "1.50.2",
                            segments: [
                                Number(
                                    1,
                                ),
                                Number(
                                    50,
                                ),
                                Number(
                                    2,
                                ),
                            ],
                        },
                    },
                ],
            },
            dep_type: Development,
        },
    ],
    metadata: {},
    platform: Ruby,
    bindir: "bin",
    post_install_message: None,
    requirements: [],
    required_ruby_version: Requirement {
        constraints: [
            VersionConstraint {
                operator: Pessimistic,
                version: Version {
                    version: "3.0",
                    segments: [
                        Number(
                            3,
                        ),
                        Number(
                            0,
                        ),
                    ],
                },
            },
        ],
    },
    required_rubygems_version: Requirement {
        constraints: [
            VersionConstraint {
                operator: Pessimistic,
                version: Version {
                    version: "3.3",
                    segments: [
                        Number(
                            3,
                        ),
                        Number(
                            3,
                        ),
                    ],
                },
            },
        ],
    },
    test_files: [],
    extra_rdoc_files: [],
    rdoc_options: [],
    cert_chain: [],
    signing_key: None,
    autorequire: None,
    installed_by_version: None,
}
